gobblin-core/src/main/java/gobblin/instrumented/writer/InstrumentedDataWriterBase.java

```diff
@@ -170,10 +198,84 @@ public State getFinalState() {
   public void close()
       throws IOException {
     this.closer.close();
+    if (this.writerMetricsUpdaterFuture.isPresent()) {
+      this.writerMetricsUpdaterFuture.get().cancel(false);
```

Comment:
Will this also shutdown the thread? We had issues were non-shutdown threads would block Gobblin from exiting.